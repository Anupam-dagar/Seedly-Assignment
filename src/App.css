body {
  font-size: 0.9rem;
}

/* Navigation Bar */
.bg-nav {
  background: white;
}

.bg-input {
  background-color: #f8f9fa !important;
  border: 0 !important;
}

.button-search {
  border-radius: 18px;
  border: 2px solid #286fdc;
}

.nav-fontsize {
  font-size: 12px;
}

.text-navbar {
  color: #121416 !important;
}

.text-navbar:hover {
  color: #286fdc !important;
}

.button-search:hover {
  background-color: white;
  border-radius: 18px;
  border: 2px solid #286fdc;
  color: #286fdc;
}

.ig-size {
  width: 13rem !important;
}

.ig-size-expanded {
  width: 22rem !important;
}

.navbar-toggler-icon {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='30' height='30' viewBox='0 0 30 30'%3e%3cpath stroke='rgba(40, 11, 220, 1.0)' stroke-linecap='round' stroke-miterlimit='10' stroke-width='2' d='M4 7h22M4 15h22M4 23h22'/%3e%3c/svg%3e") !important;
}

.navbar-toggler {
  border: 0 !important;
}

/* Header */
.btn-header {
  background-color: white;
  border-radius: 45px;
  color: #286fdc;
}

/* Sidebar */

.text-sidebar {
  color: #343a40 !important;
}

.text-sidebar:hover {
  color: #286fdc !important;
}

.text-sidebar.active {
  color: #286fdc !important;
  font-weight: bold;
  background-color: rgb(244, 245, 246);
  border-radius: 4px;
}

.card-title.h5 {
  font-size: 1rem !important;
}

/* Question Feed Topbar */
.qf-topbar {
  color: #6c757d !important;
  font-weight: bold;
}

.qf-topbar.active {
  color: #286fdc !important;
}

.qf-label {
  background-color: #f4f5f6;
  color: #9aa1b3;
  border: 0;
  font-size: 12px;
}

.qf-label:hover {
  color: #286fdc;
  background-color: #ddeaff;
}

/* Question Feed Follow Share Bar*/
.btn-follow {
  background-color: white;
  border: 1px solid #dee3f0;
  color: #40485a;
  font-size: 12px;
}

.btn-follow:hover {
  background-color: white;
  border: 1px solid #dee3f0;
  color: #40485a;
  font-size: 12px;
}

/* Answer Top Bar */
.author {
  font-size: 12px;
}

/* Like Button */
.btn-like {
  position: absolute;
  right: 20px;
  bottom: 5px;
  background-color: white;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 4px 0px;
  border: 1px solid rgb(244, 245, 246);
  border-radius: 30px;
}

.btn-like:hover {
  background-color: white;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 4px 0px;
  border: 1px solid rgb(244, 245, 246);
  border-radius: 30px;
}

/* Answer Button */
.btn-answer {
  width: 100%;
  color: rgb(87, 207, 141);
  background-color: rgb(243, 252, 248);
  border: 0;
}

.btn-answer:hover {
  width: 100%;
  color: rgb(87, 207, 141);
  background-color: rgb(243, 252, 248);
  border: 0;
}
