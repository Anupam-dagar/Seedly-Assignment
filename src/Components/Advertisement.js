import React from "react";
import { Card, Container } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

function Advertisement(props) {
  return (
    <Card style={{ border: 0, width: "13rem" }}>
      <Card.Body>
        <Card.Title>Advertisement</Card.Title>
        <Container className="m-4">
          <FontAwesomeIcon icon={["fas", "bullhorn"]} size="5x" />
        </Container>
        <Card.Text>Advertisement runs here</Card.Text>
      </Card.Body>
    </Card>
  );
}

export default Advertisement;
