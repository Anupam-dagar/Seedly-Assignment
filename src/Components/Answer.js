import React, { Component } from "react";
import { Image, Media } from "react-bootstrap";
import author from "../assets/author.png";

class Answer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      content: this.props.content,
    };
  }

  componentDidUpdate(earlierProps) {
    if (earlierProps.content !== this.props.content) {
      this.setState({ content: this.props.content });
    }
  }

  render() {
    return (
      <Media>
        <Media.Body className="mr-3">
          <p className="text-justify text-break">{this.state.content}</p>
        </Media.Body>
        <Image src={author} width={60} rounded />
      </Media>
    );
  }
}

export default Answer;
