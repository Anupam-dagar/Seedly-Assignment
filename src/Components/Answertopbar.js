import React, { Component } from "react";
import { Image, Media } from "react-bootstrap";
import author from "../assets/author.png";

class AnswerTopBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: this.props.user,
      time: this.props.time
    };
  }

  render() {
    return (
      <Media>
        <Image src={author} className="mr-3" width={35} rounded />
        <Media.Body>
          <div className="author">
            {this.state.user.name}
            {this.state.user.designation !== "" &&
              `, ${this.state.user.designation}`}
          </div>
          <span className="text-primary font-weight-bold">{this.state.user.level.name}</span>{" "}
          <span className="ml-1 text-secondary">Updated {this.state.time}h ago</span>
        </Media.Body>
      </Media>
    );
  }
}

export default AnswerTopBar;
