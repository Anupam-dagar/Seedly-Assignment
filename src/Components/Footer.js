import React, { Component } from "react";
import { Image } from "react-bootstrap";
import appstore from "../assets/app-store.png";
import playstore from "../assets/google-play.png";
import FooterIcons from "./Footericons";

// Footer Component
export class Footer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: [
        {
          heading: "Company",
          tags: ["About", "Contact", "Careers", "Privacy", "Terms"],
        },
        {
          heading: "Product Reviews",
          tags: [
            "Credit Cards",
            "Electricity Retailers",
            "Mobile Plans",
            "Online Brokerages",
            "P2P Lending",
            "Robo-Advisors",
            "Savings Accounts",
            "Travel Insurance",
          ],
        },
        {
          heading: "Quick Tools",
          tags: ["Savings Account Calculator", "Levels and Badges"],
        },
        {
          heading: "Seedly App",
          tags: ["App Reviews", "FAQ", "Security"],
        },
      ],
    };
  }

  render() {
    return (
      <footer id="sticky-footer" className="text-white">
        <div className="p-4 bg-secondary">
          <div className="d-flex justify-content-around">
            {this.state.data.map((data, index) => (
              <div key={index} className="d-flex flex-column mb-3">
                <div className="p-1 font-weight-bold">{data.heading}</div>
                {data.tags.map((tag, tagIndex) => (
                  <div key={tagIndex} className="p-1">
                    {tag}
                  </div>
                ))}
              </div>
            ))}
            <div className="d-flex flex-column mb-3">
              <div className="p-1 font-weight-bold">Download the app</div>
              <div className="p-1">
                <Image src={appstore} width={100} />
              </div>
              <div className="p-1">
                <Image src={playstore} width={100} />
              </div>
            </div>
          </div>
        </div>

        <FooterIcons />
      </footer>
    );
  }
}

export default Footer;
