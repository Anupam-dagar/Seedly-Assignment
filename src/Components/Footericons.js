import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

// FooterIcons Component
function FooterIcons() {
  return (
    <div className="bg-dark">
      <div className="d-flex ml-4 p-4">
        <FontAwesomeIcon
          className="ml-4 mr-4"
          icon={["fab", "twitter"]}
          size="2x"
        />
        <FontAwesomeIcon
          className="ml-4 mr-4"
          icon={["fab", "instagram"]}
          size="2x"
        />
        <FontAwesomeIcon
          className="ml-4 mr-4"
          icon={["fab", "facebook"]}
          size="2x"
        />
        <FontAwesomeIcon
          className="ml-4 mr-4"
          icon={["fab", "telegram-plane"]}
          size="2x"
        />
      </div>
    </div>
  );
}

export default FooterIcons;
