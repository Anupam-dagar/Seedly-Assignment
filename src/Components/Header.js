import React from "react";
import { Jumbotron, Button } from "react-bootstrap";
import headerImage from "../assets/header.svg";

// Header Component
function Header() {
  return (
    <Jumbotron
      fluid
      className="mx-auto mb-0"
      style={{
        backgroundImage: `url(${headerImage})`,
        backgroundSize: "cover",
        backgroundRepeat: "no-repeat",
        backgroundPosition: "center center",
        minHeight: "18rem",
      }}
    >
      <div className="d-flex justify-content-center align-items-center">
        <div className="bd-highlight text-light">
          <h2>Let's Talk Finance</h2>
        </div>
      </div>
      <div className="d-flex justify-content-center align-items-center">
        <div className="bd-highlight text-light text-center">
          Ask for opinions and get answers from other Singaporeans.
        </div>
      </div>
      <div className="d-flex justify-content-center align-items-center">
        <div className="p-3 bd-highlight text-light">
          <Button className="btn-header font-weight-bold pl-6 pr-6 pt-2 pb-2">
            ASK A QUESTION
          </Button>
        </div>
      </div>
    </Jumbotron>
  );
}

export default Header;
