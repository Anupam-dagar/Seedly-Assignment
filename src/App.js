import React from "react";
import NavigationBar from "./Components/NavigationBar";
import Header from "./Components/Header";
import Footer from "./Components/Footer";
import Sidebar from "./Components/Sidebar";
import Advertisement from "./Components/Advertisement";
import QuestionFeed from "./Components/Questionfeed";
import { Jumbotron, Container } from "react-bootstrap";
import "./style.scss";
import "./App.css";

function App() {
  return (
    <>
      <NavigationBar />
      <Header />
      <Jumbotron fluid className="mb-0 pt-3">
        <Container>
          <div className="d-flex justify-content-center">
            <div className="d-flex flex-column">
              <Sidebar />
            </div>
            <div className="ml-3 flex-fill">
              <QuestionFeed />
            </div>
            <div className="ml-3 d-none d-lg-block">
              <Advertisement />
            </div>
          </div>
        </Container>
      </Jumbotron>
      <Footer />
    </>
  );
}

export default App;
