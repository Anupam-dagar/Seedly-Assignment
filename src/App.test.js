import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import { Advertisement } from "./Components/Advertisement";
import { Answer } from "./Components/Answer";
import { AnswerTopBar } from "./Components/Answertopbar";
import { Footer } from "./Components/Footer";
import { FooterIcons } from "./Components/Footericons";
import { Header } from "./Components/Header";
import { NavigationBar } from "./Components/NavigationBar";
import { NavItem } from "./Components/Navitem";
import { QuestionAnswerCard } from "./Components/Questionanswercard";
import { QuestionCard } from "./Components/Questioncard";
import { QuestionCardTopics } from "./Components/Questioncardtopics";
import { QuestionFeed } from "./Components/Questionfeed";
import { QuestionFeedTopbar } from "./Components/Questionfeedtopbar";
import { QuestionFollowShareBar } from "./Components/Questionfollowsharebar";
import { Sidebar } from "./Components/Sidebar";
import { BrowserRouter } from "react-router-dom";

it("should render App component", () => {
  const component = shallow(<App />);

  expect(component).toMatchSnapshot();
});

it("should render Navigation Bar component", () => {
  const component = shallow(<NavigationBar />);

  expect(component).toMatchSnapshot();
});

it("should render Advertisement component", () => {
  const component = shallow(<Advertisement />);

  expect(component).toMatchSnapshot();
});

it("should render Answer component", () => {
  const component = shallow(<Answer />);

  expect(component).toMatchSnapshot();
});

it("should render AnswerTopBar component", () => {
  const user = {
    id: 1,
    name: "Jennings Parisian",
    designation: "Investor Factors Developer",
    level: {
      id: 4,
      name: "Level 4",
    },
  };
  const time = 12;
  const component = shallow(<AnswerTopBar user={user} time={time} />);

  expect(component).toMatchSnapshot();
});

it("should render Footer component", () => {
  const component = shallow(<Footer />);

  expect(component).toMatchSnapshot();
});

it("should render FooterIcons component", () => {
  const component = shallow(<FooterIcons />);

  expect(component).toMatchSnapshot();
});

it("should render Header component", () => {
  const component = shallow(<Header />);

  expect(component).toMatchSnapshot();
});

it("should render NavItem component", () => {
  const component = shallow(<NavItem />);

  expect(component).toMatchSnapshot();
});

it("should render QuestionAnswerCard component", () => {
  const answer = {
    id: 186,
    content:
      "Eius consectetur consequatur harum molestiae nulla delectus omnis quidem molestiae. Error ut ex tempora provident. Voluptatem ea perferendis occaecati rerum atque.\n \rNemo quos modi. Earum tempore sint eligendi unde atque dolorum. Ab blanditiis qui ut illum.\n \rVoluptatem eum doloremque. Exercitationem optio dolorum quaerat occaecati voluptas vero error. Quis sunt blanditiis laborum quam qui expedita quos. Est molestias sint dignissimos consequatur inventore explicabo.",
    questionId: 44,
    user: {
      id: 6,
      name: "Aisha Powlowski",
      designation: "Customer Security Supervisor",
      level: {
        id: 4,
        name: "Level 4",
      },
    },
    likeCount: 0,
    answerUpdated: 4,
  };
  const component = shallow(<QuestionAnswerCard answer={answer} />);

  expect(component).toMatchSnapshot();
});

it("should render QuestionCard component", () => {
  const question = {
    id: 14,
    title:
      "Nisi nihil tempora commodi eligendi excepturi exercitationem. Illum fuga sint. Doloremque iste voluptatem velit occaecati ea voluptas omnis.",
    followers: 0,
    isTrending: true,
    isUnanswered: false,
    topics: [
      {
        id: 1,
        name: "haptic",
        isFeatured: false,
      },
      {
        id: 8,
        name: "Connecticut",
        isFeatured: false,
      },
      {
        id: 17,
        name: "Visionary",
        isFeatured: true,
      },
      {
        id: 22,
        name: "generate",
        isFeatured: false,
      },
      {
        id: 50,
        name: "Avon",
        isFeatured: false,
      },
      {
        id: 24,
        name: "Hills",
        isFeatured: true,
      },
      {
        id: 19,
        name: "payment",
        isFeatured: true,
      },
      {
        id: 50,
        name: "Avon",
        isFeatured: false,
      },
      {
        id: 20,
        name: "Licensed",
        isFeatured: true,
      },
    ],
    answers: [
      {
        id: 4,
        content:
          "Id voluptates magni. Molestias iusto quo saepe quia deleniti inventore ut non modi. Quas quo et. Aut saepe dolor dignissimos et dolor cumque maxime repellendus animi. Voluptatum repellat autem voluptatibus quis voluptates. Quos rem voluptatem voluptatem earum consequatur sint.\n \rNon voluptas dolorum cumque dolores voluptas fugit fugit pariatur dolorem. Velit iste maiores libero placeat saepe. Eveniet repellendus a recusandae similique quis molestiae sit sint. Dolorum aut consequuntur quod laborum autem fugiat.\n \rQuidem ut eius delectus unde velit iure sed tempore. Ut distinctio enim quod maiores. Est et ea voluptatem rerum. Iure explicabo distinctio nihil. Ipsam similique ratione est.",
        questionId: 14,
        user: {
          id: 3,
          name: "Madisen Brekke",
          designation: "Senior Web Designer",
          level: {
            id: 5,
            name: "Level 5",
          },
        },
        likeCount: 4,
        answerUpdated: 1,
      },
    ],
  };
  const component = shallow(<QuestionCard questionData={question} />);

  expect(component).toMatchSnapshot();
});

it("should render QuestionCardTopics component", () => {
  const topics = [
    {
      id: 2,
      name: "calculating",
      isFeatured: true,
    },
    {
      id: 1,
      name: "haptic",
      isFeatured: false,
    },
    {
      id: 19,
      name: "payment",
      isFeatured: true,
    },
  ];
  const component = shallow(<QuestionCardTopics topics={topics} />);

  expect(component).toMatchSnapshot();
});

it("should render QuestionFeed component", () => {
  const questions = [
    {
      id: 76,
      title:
        "Voluptates natus ut commodi nisi temporibus veniam quisquam omnis. Non minima ab. Officia nisi voluptas ut aut rerum ut commodi accusantium eius. Et quibusdam sed temporibus qui. Minima alias fugit ipsam optio accusantium adipisci ullam voluptates ut.",
      followers: 7,
      isTrending: false,
      isUnanswered: false,
      topics: [
        {
          id: 2,
          name: "calculating",
          isFeatured: true,
        },
        {
          id: 1,
          name: "haptic",
          isFeatured: false,
        },
        {
          id: 19,
          name: "payment",
          isFeatured: true,
        },
      ],
      answers: [
        {
          id: 114,
          content:
            "Impedit architecto adipisci ea veniam dolores eligendi commodi sint. Reiciendis cupiditate id pariatur perferendis. Dolore adipisci rerum voluptatem tenetur et ut. Non magnam vitae sit ut. Officia laboriosam dignissimos unde corporis cumque error ut minima asperiores.\n \rNumquam repellendus nam. Nihil soluta temporibus culpa. Et voluptas ea qui impedit. Doloribus qui qui.\n \rEt architecto nemo officiis soluta nulla eos. Nostrum sed minima expedita optio qui sint laboriosam quibusdam. Odit voluptatem ullam quo aut nobis vel suscipit.",
          questionId: 76,
          user: {
            id: 2,
            name: "Everett D'Amore",
            designation: "Forward Optimization Designer",
            level: {
              id: 4,
              name: "Level 4",
            },
          },
          likeCount: 7,
          answerUpdated: 1,
        },
        {
          id: 127,
          content:
            "Et autem voluptatem iusto vero quia quia. Eum eum velit laudantium quod. Occaecati illo quia.\n \rVel et itaque quos voluptate est repellat nihil officiis. Voluptas odit sint inventore accusamus sit delectus. Porro animi iste repudiandae voluptatem iure ut natus. Et cupiditate nostrum soluta optio nihil odit voluptatem. Molestias quaerat sunt rerum voluptatem in est enim doloribus eveniet.\n \rExcepturi culpa velit. Nihil repellat nostrum mollitia aut. Atque dolorem voluptas eveniet. Veniam ducimus voluptatum dicta error ut. Voluptatem rerum ut accusamus omnis.",
          questionId: 76,
          user: {
            id: 5,
            name: "Samantha Mayert",
            designation: "Regional Solutions Facilitator",
            level: {
              id: 5,
              name: "Level 5",
            },
          },
          likeCount: 2,
          answerUpdated: 14,
        },
        {
          id: 168,
          content:
            "Voluptas sunt qui exercitationem non alias. Molestias at iure sed et id fuga officiis rerum. Similique est tenetur. Qui necessitatibus eos aspernatur sint nisi asperiores veritatis sapiente dignissimos.\n \rMinus excepturi quis dolores. Aut et at enim rerum sit quo non qui et. Mollitia placeat et veniam fugiat. Cumque explicabo distinctio enim labore quidem sit saepe ut quasi. Non est inventore et consequuntur.\n \rQui exercitationem non illum architecto voluptatem ut. Reiciendis eum repudiandae exercitationem tempore quia ut illum. Beatae ut id. Magni et blanditiis fuga dolorem distinctio voluptate voluptas dolorum. Voluptatem nobis voluptatem tenetur doloribus aut recusandae quaerat. Ut impedit ut alias tenetur.",
          questionId: 76,
          user: {
            id: 3,
            name: "Madisen Brekke",
            designation: "Senior Web Designer",
            level: {
              id: 5,
              name: "Level 5",
            },
          },
          likeCount: 1,
          answerUpdated: 2,
        },
      ],
    },
    {
      id: 4,
      title:
        "Est sunt commodi. Reiciendis alias nam ut consequatur alias maiores temporibus ut culpa.",
      followers: 0,
      isTrending: false,
      isUnanswered: true,
      topics: [
        {
          id: 2,
          name: "calculating",
          isFeatured: true,
        },
        {
          id: 43,
          name: "paradigms",
          isFeatured: false,
        },
        {
          id: 14,
          name: "platforms",
          isFeatured: true,
        },
        {
          id: 3,
          name: "back-end",
          isFeatured: true,
        },
        {
          id: 36,
          name: "SMTP",
          isFeatured: false,
        },
        {
          id: 46,
          name: "Ball",
          isFeatured: true,
        },
        {
          id: 43,
          name: "paradigms",
          isFeatured: false,
        },
        {
          id: 1,
          name: "haptic",
          isFeatured: false,
        },
      ],
      answers: [],
    },
  ];
  const component = shallow(<QuestionFeed questions={questions} />);

  expect(component).toMatchSnapshot();
});

it("should render QuestionFeedTopBar component", () => {
  const component = shallow(<QuestionFeedTopbar />);

  expect(component).toMatchSnapshot();
});

it("should render QuestionFollowShareBar component", () => {
  const component = shallow(<QuestionFollowShareBar />);

  expect(component).toMatchSnapshot();
});

it("should render Sidebar component", () => {
  const component = shallow(
    <BrowserRouter>
      <Sidebar />
    </BrowserRouter>
  );

  expect(component).toMatchSnapshot();
});
