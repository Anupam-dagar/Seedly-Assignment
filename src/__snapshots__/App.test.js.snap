// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Advertisement component 1`] = `ShallowWrapper {}`;

exports[`should render Answer component 1`] = `ShallowWrapper {}`;

exports[`should render AnswerTopBar component 1`] = `ShallowWrapper {}`;

exports[`should render App component 1`] = `ShallowWrapper {}`;

exports[`should render Footer component 1`] = `ShallowWrapper {}`;

exports[`should render FooterIcons component 1`] = `ShallowWrapper {}`;

exports[`should render Header component 1`] = `ShallowWrapper {}`;

exports[`should render NavItem component 1`] = `ShallowWrapper {}`;

exports[`should render Navigation Bar component 1`] = `ShallowWrapper {}`;

exports[`should render QuestionAnswerCard component 1`] = `ShallowWrapper {}`;

exports[`should render QuestionCard component 1`] = `ShallowWrapper {}`;

exports[`should render QuestionCardTopics component 1`] = `ShallowWrapper {}`;

exports[`should render QuestionFeed component 1`] = `ShallowWrapper {}`;

exports[`should render QuestionFeedTopBar component 1`] = `ShallowWrapper {}`;

exports[`should render QuestionFollowShareBar component 1`] = `ShallowWrapper {}`;

exports[`should render Sidebar component 1`] = `ShallowWrapper {}`;
